src/common-subexpr-elim.rkt

```diff
@@ -77,8 +72,16 @@
       (define edges* (filter-no-edges deps*))
       (loop (cons chosen sorted) deps* edges*)])))
 
-;; eliminator
-(define (common-subexpr-elim vars expr)
+(define (dict-set** dict keys vals)
+  (for/fold ([dict dict]) ([k keys] [v vals])
+    (dict-set dict k v)))
+
+(define (hash-set**! h keys vals)
+  (for ([k keys] [v vals])
+    (hash-set! h k v)))
+
+;; main cse procedure
+(define (common-subexpr-elim vars props expr)
```

Comment:
This is kind of unreviewable, but I skimmed parts of it and it looks fine.

Reply:
Its been a while since I touched this code and I'm having trouble reading it, so that's not a good sign. I'll go through and try to comment a lot of stuff.